src/server/pps/server/api_server.go

```diff
@@ -2194,37 +2103,41 @@ func (a *apiServer) CreatePipelineInTransaction(
 
 	// Create/update output branch (creating new output commit for the pipeline
 	// and restarting the pipeline)
-	if err := a.env.PfsServer().CreateBranchInTransaction(txnCtx, &pfs.CreateBranchRequest{
-		Branch:     outputBranch,
-		Provenance: provenance,
-		Head:       outputBranchHead,
-	}); err != nil {
-		return errors.Wrapf(err, "could not create/update output branch")
+	if update && request.Reprocess {
```

Comment:
This condition should be negated and just be a single block with what is currently in the else block.

Reply:
How are we breaking the commit chain if we don't update the branch in the update && reprocess case?

Reply:
We do not break the commit chain, is that necessary?

Reply:
Ohh, that is right. We don't need to break the chain anymore because we will perform the correct deletions on the parent to remove the old state.